Library/MapleBacon/MapleBacon/ImageDownloadOperation.swift

```diff
@@ -7,113 +7,117 @@ import UIKit
 public typealias ImageDownloaderCompletion = (ImageInstance?, NSError?) -> Void
 
 public protocol ImageDownloaderDelegate {
-    func imageDownloaderDelegate(downloader: ImageDownloadOperation, didReportProgress progress: NSProgress);
+    func imageDownloaderDelegate(downloader: ImageDownloadOperation, didReportProgress progress: Progress)
 }
 
-public class ImageDownloadOperation: NSOperation {
+public class ImageDownloadOperation: Operation {
 
-    private var imageURL: NSURL
+    private var imageURL: URL
     private var delegate: ImageDownloaderDelegate?
-    private var session: NSURLSession?
-    private var task: NSURLSessionDownloadTask?
-    private var resumeData: NSData?
-    private let progress: NSProgress = NSProgress()
+    private var session: Foundation.URLSession?
+    private var task: URLSessionDownloadTask?
+    private var resumeData: Data?
+    private let progress: Progress = Progress()
 
     public var completionHandler: ImageDownloaderCompletion?
 
-    public convenience init(imageURL: NSURL) {
+    private var _finished = false {
+        didSet {
+            didChangeValue(forKey: "isFinished")
+        }
+        willSet {
+           willChangeValue(forKey: "isFinished")
+        }
+    }
+    
+    override public var isFinished: Bool {
+        get {
+            return _finished
+        }
+    }
+    
+    public convenience init(imageURL: URL) {
         self.init(imageURL: imageURL, delegate: nil)
     }
 
-    public init(imageURL: NSURL, delegate: ImageDownloaderDelegate?) {
+    public init(imageURL: URL, delegate: ImageDownloaderDelegate?) {
         self.imageURL = imageURL
         self.delegate = delegate
         super.init()
     }
 
     public override func start() {
-        let sessionConfiguration = NSURLSessionConfiguration.defaultSessionConfiguration()
-        session = NSURLSession(configuration: sessionConfiguration, delegate: self,
-                delegateQueue: NSOperationQueue.mainQueue())
+        let sessionConfiguration = URLSessionConfiguration.default()
+        session = Foundation.URLSession(configuration: sessionConfiguration, delegate: self,
```

Comment:
Actually, can let the compiler infer a bunch of stuff here and shorten the whole expression to:

```swift
session = URLSession(configuration: .default(), delegate: self, delegateQueue: .main())
```